cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/EOGWorklist.kt

```diff
@@ -248,9 +248,18 @@ inline fun <reified K : Node, V> iterateEOG(
     while (worklist.isNotEmpty()) {
         val (nextNode, state) = worklist.pop()
 
-        val newState = transformation(nextNode, state.duplicate(), worklist)
+        val insideBB =
```

Comment:
This should check if we're not near the beginning/end of a basic block (i.e., there are no merge points or branches of the EOG nearby). If that's the case, we just parse the whole basic block and do not want to duplicate the state. Near the beginning/end, we do want to copy the state to avoid terminating the iteration too early by messing up with the state-changing checks.

Reply:
Can you maybe add this as a comment to the variable. Otherwise it is confusing because we don’t actually do any check if we are in a compound statement or similar but it’s purely based on the “entry” within the block.